src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -86,10 +88,11 @@ DecisionTree<FitnessFunction,
 
   // Set the correct dimensionality for the dimension selector.
   dimensionSelector.Dimensions() = tmpData.n_rows;
+  arma::Row<size_t> passlabels = tmpLabels;
```

Comment:
Previously `tmbLabels` was causing a failure 
```
cannot convert ‘tmplabels’ (type ‘truelabelstype {aka arma::subview_row<long unsigned int>}’) to type ‘arma::row<long unsigned int>&’
```
So, I changed it to remove this failure.

Reply:
Maybe I missed something but:

https://github.com/mlpack/mlpack/blob/ab3f7a11788d3d5a2b9e1cabd58e0258dc70dcec/src/mlpack/methods/decision_tree/decision_tree_impl.hpp#L658-L660

looks really similar, which doesn't create a copy first. Maybe I missed something.

Reply:
I removed this to show the failure. I think it is failing because of adaboost_test. I can't able to understand what is the main cause of failure. @zoq Please have a look when build get completed

Reply:
Thanks, looks like there is some issue with the type deduction, from subview (`.row(0)`) to the actual label type. Will have to take a closer look into the issue.

Reply:
@zoq I think this error is because of `adaboost_test` because we are using `labels.row(0)` 
```
  AdaBoost<ID3DecisionStump > a(inputData, labels.row(0), numClasses, ds,
      iterations, tolerance);
``` 
I will try any different way to do this.